### PR TITLE
Remove AWS Active Tracing span linking

### PR DESCRIPTION
**Pull request by tylerbenson**
This aligns with the latest spec change:
https://github.com/open-telemetry/semantic-conventions/pull/354

Providing the Lambda span's parent context should continue to be the responsibility of the global propagator.

Related: https://github.com/open-telemetry/opentelemetry-java-contrib/pull/1032

### CODE REVIEW

File: instrumentation/aws-lambda/aws-lambda-core-1.0/testing/src/main/java/io/opentelemetry/instrumentation/awslambdacore/v1_0/AbstractAwsLambdaTest.java [hunk 1]
```diff
@@ -90,39 +87,4 @@ void handlerTracedWithException() {
...
-    assertThat(result).isEqualTo("world");
-
-    testing()
-        .waitAndAssertTraces(
-            trace ->
-                trace.hasSpansSatisfyingExactly(
-                    span ->
-                        span.hasName("my_function")
-                            .hasKind(SpanKind.SERVER)
-                            .hasLinksSatisfying(
```
Comment (laurit) [inline instrumentation/aws-lambda/aws-lambda-core-1.0/testing/src/main/java/io/opentelemetry/instrumentation/awslambdacore/v1_0/AbstractAwsLambdaTest.java:109]: would it make sense to keep rest of the test except the `hasLinksSatisfying` part?

Comment (tylerbenson) [reply instrumentation/aws-lambda/aws-lambda-core-1.0/testing/src/main/java/io/opentelemetry/instrumentation/awslambdacore/v1_0/AbstractAwsLambdaTest.java:109]: I thought I added this test when I added the span link. It was actually added before, so I've updated the test instead now.

**Review by laurit — APPROVED** [review]

**Comment by tylerbenson** [comment]
FYI @rapphil

**Comment by trask** [comment]
cc @wangzlei @srprash

**Comment by tylerbenson** [comment]
@trask @laurit please remember to do a release of the contrib repo before the next javaagent release.  It doesn't look like that has a regular release cadence like the other java repos do.

**Comment by trask** [comment]
> @trask @laurit please remember to do a release of the contrib repo before the next javaagent release. It doesn't look like that has a regular release cadence like the other java repos do.

thanks for the reminder, sometimes we miss releasing from that repo, just released contrib [v1.34.0](https://github.com/open-telemetry/opentelemetry-java-contrib/releases/tag/v1.34.0)

**Comment by trask** [comment]
I think if you rebase (and pull in #10969) it should fix the license check

**Comment by tylerbenson** [comment]
Any more blockers here?

**Comment by trask** [comment]
we'll merge it before `v2.3.0`. would be great to get an approval from someone in the faas sig if possible.

**Review by rapphil — APPROVED** [review]

**Pull request merged**